cmd/stack_create.go

```diff
@@ -0,0 +1,182 @@
+// Copyright © 2019 IBM Corporation and others.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"archive/zip"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+type stackCreateCommandConfig struct {
+	*RootCommandConfig
+	copy string
+}
+
+func newStackCreateCmd(rootConfig *RootCommandConfig) *cobra.Command {
+	config := &stackCreateCommandConfig{RootCommandConfig: rootConfig}
+
+	var stackCmd = &cobra.Command{
+		Use:   "create <name>",
+		Short: "Create a new stack as a copy of an existing stack",
+		Long: `This command will create a new stack as a copy of an existing sample stack in the current directory that has the structure of an Appsody stack.
+        
+		If a copy flag is specified, stack create command will create a new stack as a copy of that existing stack.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) < 1 {
+				return errors.New("Required parameter missing. You must specify a stack name")
+			}
+
+			stack := args[0]
+
+			match, err := validateKubernetesResourceName(stack)
+
+			if !match {
+				return err
+			}
+			exists, err := Exists(stack)
+
+			if err != nil {
+				return err
+			}
+
+			if exists {
+				return errors.New("A stack named" + stack + "already exists in your directory. Specify a unique stack name")
+			}
+
+			err = downloadFileToDisk("https://github.com/appsody/stacks/archive/master.zip", filepath.Join(getHome(rootConfig), "extract", "repo.zip"), config.Dryrun)
+			if err != nil {
+				return err
+			}
+			_, projectType, err := parseProjectParm(config.copy, config.RootCommandConfig)
+			if err != nil {
+				return err
+			}
+
+			valid, unzipErr := unzip(filepath.Join(getHome(rootConfig), "extract", "repo.zip"), stack, config.copy)
+
+			if unzipErr != nil {
+				return unzipErr
+			}
+
+			if !valid {
+				return errors.Errorf("Invalid stack name:" + stack + ". Stack name must be in the format <repo>/<stack>")
```

Comment:
Instead of `stack`, it should be `config.copy`. 
I get the following error:
![image](https://user-images.githubusercontent.com/30461857/66998473-3c77f580-f0cc-11e9-9465-e5fe814bbd1f.png)

Reply:
Yesterday I updated the error messages which @helenmasters recommended, and I accidentally wrote stack instead of copy.config. Thanks for testing